addons/HotelPvP/scripts/Game/AFM_HotelPvP/GameMode/AFM_FactoryPvPGameMode.c

```diff
@@ -0,0 +1,442 @@
+			foreach(SCR_SpawnPoint sp: spawnPoints)
+			{
+				sp.SetSpawnPointEnabled_S(!sp.IsSpawnPointEnabled());
+			}
+		}
+		OnMatchSituationChanged();
+	}
+	
+	void GameEndCheck() 
+	{
+		if (!m_bHasGameStarted)
+			return;
+		
+		m_iAttackersRemaining = GetFactionRemainingPlayers(m_sAttackerFactionKey);
+		m_iDefendersRemaining = GetFactionRemainingPlayers(m_sDefenderFactionKey);
+		
+		bool attackersDead = m_iAttackersRemaining == 0;
+		bool defendersDead = m_iDefendersRemaining == 0;
+		bool hostageStatus = AllHostagesFreed();
+		
+		PrintFormat("GameEndCheck: A:%1 D:%2 H:%3", attackersDead, defendersDead, hostageStatus, level:LogLevel.SPAM);
+		
+		if (AllHostagesFreed())
+		{
+			Print("All hostages escaped", LogLevel.NORMAL);
+			GameEnd(m_sAttackerFactionKey);
+			return;
+		}
+		
+		if (attackersDead && defendersDead)
+		{
+			GameEndTie();
+			return;
+		}
+		
+		if (attackersDead)
+		{
+			GameEndDefendersWin();
+			return;	
+		}
+		
+		if (defendersDead) 
+		{
+			GameEndAttackersWin();
+			return;
+		}
+	}
+	
+	protected bool AllHostagesFreed()
+	{
+		return GetGame().GetWorld().QueryEntitiesBySphere(
+				GetOrigin(),
+				m_iGameAreaRadius,
+				ProcessFoundHostages,
+				FilterHostageEntities,
+				EQueryEntitiesFlags.DYNAMIC
+		);
+	}
+	
+	protected bool ProcessFoundHostages(IEntity entity)
+	{
+		return false;
+	}
+	
+	protected bool FilterHostageEntities(IEntity entity)
+	{
+		if(entity.Type() != SCR_ChimeraCharacter)
+			return false; 
+		
+		FactionAffiliationComponent fac = FactionAffiliationComponent.Cast(entity.FindComponent(FactionAffiliationComponent));
+		return fac && fac.GetAffiliatedFactionKey() == "CIV";
+	}
+	
+	void TimeoutGameEnd()
+	{
+		if (!IsMaster())
+			return;
+		
+		PrintFormat("Timeout reached, ending game with {%1} faction win", m_TimeoutWinnerFactionKey);
+		SCR_Faction faction = SCR_Faction.Cast(m_FactionManager.GetFactionByKey(m_TimeoutWinnerFactionKey));
+		
+		
+		if (!faction)
+		{
+			GameEndTie();
+			return;
+		}
+	
+		GameEnd(m_TimeoutWinnerFactionKey);
+	}
+	
+	
+	protected void GameEnd(FactionKey fKey)
+	{
+		m_bHasGameStarted = false;
+		
+		Faction faction = m_FactionManager.GetFactionByKey(fKey);
+		int factionId = m_FactionManager.GetFactionIndex(faction);
+		SCR_GameModeEndData endData = SCR_GameModeEndData.CreateSimple(EGameOverTypes.ENDREASON_SCORELIMIT, winnerFactionId:factionId);
+		EndGameMode(endData);
+	}
+	
+	protected void GameEndTie()
+	{
+		Print("Game tie", LogLevel.NORMAL);
+		SCR_GameModeEndData endData = SCR_GameModeEndData.Create(
+			EGameOverTypes.FACTION_DRAW,
+			{}, {}
+		);
+		m_bHasGameStarted = false;
+		EndGameMode(endData);
+	}
+	
+	protected void GameEndDefendersWin()
+	{
+		Print("Defenders win!");
+		GameEnd(m_sDefenderFactionKey);
+	}
+	
+	protected void GameEndAttackersWin()
+	{
+		Print("Attackers win!");
+		GameEnd(m_sAttackerFactionKey);
+	}
+	
+	int GetFactionRemainingPlayers(FactionKey fKey)
```

Comment:
Make it return the players or rename to indicate that it returns the count.
```suggestion
	int GetFactionRemainingPlayersCount(FactionKey fKey)
```